src/mono/wasm/testassets/WasmOnAspNetCore/WasmBrowserClient/wwwroot/main.js

```diff
@@ -0,0 +1,38 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+import { dotnet, exit } from './_framework/dotnet.js'
+
+function getQueryParam(prameterName) {
+    const params = new URLSearchParams(location.search);
+    const paramValue = params.get(prameterName);
+    if (!paramValue)
+        throw new Error(`Missing query parameter '${prameterName}' in query: ${params}`);
+    return paramValue;
+}
+
+function getUrl() {
+    return window.location.protocol + "//" + window.location.host;
+}
+
+try {
+    const dotnetRuntime = await dotnet
+        .withElementOnExit()
+        .withExitCodeLogging()
+        .withExitOnUnhandledError()
+        .create();
+
+        dotnetRuntime.setModuleImports("main.js", {
+            Program: {
```

Comment:
nit: Feel free to skip. Passing all three values as main method args seems like a less code 🤷‍♂️